### DISCUSSION TITLE
Drumkit YAML python script

### DISCUSSION THREAD

**Discussion by caschy** [FundamentalFrequency/.github#38] [Software]
Hi all, thought I'd share a super simple/crude python script that creates a proper YAML for a directory of wav drum kit files.  It assume you have already created a directory in the drum_kits directory of LMN-3 and copied all wav files there.  It asks for the drum_kit directory name, what starting MIDI note you would like the first wav to be assigned to and whether to copy the YAML to the directory at the end. Again this is just to start things off for anyone looking to quickly add files. (see attached file, rename as py, copy to /home/pi/ directory, and execute with: python drumkit_yaml.py)

[drumkit_yaml.txt](https://github.com/FundamentalFrequency/.github/files/9031067/drumkit_yaml.txt)

**Comment by DCElfving** [comment]
This works great. Thanks!

**Comment by caschy** [comment]
Thanks for the feedback.

**Comment by kuffcakes** [comment]
This created the .yaml file for me but did not add any of the .wav files to it. Not sure if I did something wrong.

**Reply by caschy** [reply]
Make sure you have your wav files in the correct path. Also when it asks for the folder name, make sure you type in the name of the folder correctly.

**Reply by caschy** [reply]
For example, if I wanted to add my wav files and make a "testdrumkit", I would: "mkdir ~/.config/LMN-3/drum_kits/testdrumkit"  Then manually copy my files to that folder.  After, I'd run the python program and then when asked for the folder name, I would type in testdrumkit.  Should work.

**Reply by kuffcakes** [reply]
Thanks for the info. It was an user error on my part. The script added all the file names to the .yaml file. I'm not seeing the instruments on the LMN-3 after creating the .yaml file.

**Reply by caschy** [reply]
OK. Obviously make sure the yaml file got copied over to the folder.

**Reply by kuffcakes** [reply]
Confirmed in the OdysseyDrumKit folder. I opened the .yaml in the directory to confirm all the drum track names loaded.

**Reply by caschy** [reply]
OK. And obviously you've restarted LMN-DAW right?

**Reply by kuffcakes** [reply]
I restarted the LMN-3. Is restarting the LMN-DAW different?

**Reply by caschy** [reply]
Yeah same thing. The DAW is the application/software.  If the LMN GUI comes up at restart of the Pi, restart your Pi.  If you did that and it still doesn't work then it might have something to do with how the wav files are named? IDK.  The entire software for this is really new and so far nobody knows the ins and outs of it.

**Reply by kuffcakes** [reply]
Appreciate the help. Restarted the Pi a couple times with no luck.

Can you please give me a few examples of how you named your files?
All lower case like snare1.wav or snare_1.wav?
Can there only be 24 files per directory?

**Reply by stonepreston** [reply]
when you say its not showing up in the instruments, are you looking in the drum sampler or the plugins? They wont show up in the instruments section of the plugins menu. The files should show up as drum kits in the drum sampler. Can you show what your directory structure looks like ie what are the contents of .config/LMN-3, what are the contents of .config/LMN-3/drum_kits/yourdrumkit.

**Reply by kuffcakes** [reply]
Appreciate the help again.
The instruments weren't showing up in the drum sampler. The message: See the README for instructions on adding samples and drum kits!
The directory structure:
.config/LMN-3/drum_kits/Drum_Kit (it was named OdysseyDrumKit before)
The drums showed up after renaming the folder to Drum_Kit and running the script again.
I'm added samples as well and followed the instructions to create the .yaml file.

**Reply by kuffcakes** [reply]
I'll have to look into the samples I loaded because the ones briwil uploaded on a different post, worked for me. Glad this is slowly working out.

**Comment by caschy** [comment]
Working on another version with a GU
![Untitled](https://user-images.githubusercontent.com/18508692/179659666-66abe01a-3b07-44f0-96bb-08a99040d2c1.png)
I

**Reply by caschy** [reply]
https://github.com/caschy/LMN3-Drum-Kit-Mapper
